### PR TITLE
Clarify the purpose of having resource group in gateway

### PR DESCRIPTION
**Pull request by Chaho12**
## Description

People ask what is resource group in gateway every once in a while.
It must be confusing for new users, so I added some sentences to clarify
what resource group in gateway does.
- https://trinodb.slack.com/archives/CG9K9MX1V/p1710174886868119
- https://trinodb.slack.com/archives/CG9K9MX1V/p1714685877520219

## Release notes

(x) This is not user-visible or is docs only, and no release notes are required.
( ) Release notes are required. Please propose a release note for me.
( ) Release notes are required, with the following suggested text:

```markdown
* 
```

### CODE REVIEW

File: docs/resource-groups-api.md [hunk 1]
```diff
@@ -34,6 +34,13 @@
 
 # Resource groups API
 
+Resource Groups facilitates the saving and management of the resource group
```
Comment (mosabua) [inline docs/resource-groups-api.md:37]:
```suggestion
Resource groups facilitate the configuration of the Trino resource group
```

File: docs/resource-groups-api.md [hunk 2]
```diff
@@ -34,6 +34,13 @@
...
+required for the [Database Resource Group Manager](https://trino.io/docs/current/admin/resource-groups.html#database-resource-group-manager).
```
Comment (mosabua) [inline docs/resource-groups-api.md:38]:
```suggestion
using the [database resource group manager](https://trino.io/docs/current/admin/resource-groups.html#database-resource-group-manager), configured in the Trino Gateway database.
```

File: docs/resource-groups-api.md [hunk 3]
```diff
@@ -34,6 +34,13 @@
...
+Therefore, in Trino, you would configure the database endpoint for the
```
Comment (mosabua) [inline docs/resource-groups-api.md:39]: Not sure what you want to say here .. I think

```
Configure the resource groups in each Trino backend cluster to use the resource group configuration in the Trino Gateway backend database.
```

**Review by mosabua — CHANGES_REQUESTED** [review]
Have a look at my suggestions .. does that make sense? 

Maybe we add some more info from https://trinodb.slack.com/archives/CG9K9MX1V/p1714685877520219

**Review by alaturqua — CHANGES_REQUESTED** [review]
Hi, thanks for the PR. Following would be my suggestion:

Trino Gateway facilitates the management of resource groups across multiple Trino clusters. It allows users to connect their Trino clusters to a centralized resource groups table maintained in the Trino Gateway's backend database. This setup enables management of resource groups either through the [API](https://github.com/trinodb/trino-gateway/blob/main/docs/resource-groups-api.md#resource-groups-api) or the Trino Gateway user interface.

Trino offers both [file-based](https://trino.io/docs/current/admin/resource-groups.html) and [database resource group management](https://trino.io/docs/current/admin/resource-groups.html#database-resource-group-manager). Using Trino Gateway with database management avoids the need for cluster redeployments when updating resource groups, enhancing operational efficiency.

**Comment by Chaho12** [comment]
PTAL @mosabua @alaturqua :)

**Comment by alaturqua** [comment]
@Chaho12 Thanks. It looks good for me.

**Review by alaturqua — APPROVED** [review]

**Review by mosabua — APPROVED** [review]
Looks good now. Thank you.

**Pull request merged**